### DISCUSSION TITLE
Proper example of the Azure API usage

### DISCUSSION THREAD

**Discussion by orthd** [explosion/spacy-llm#350] [Q&A] [answered]
Hi spacy-llm community,

I tried recently to use the GPT model deployed in  **Azure OpenAI** from the spacy-llm to do, e.g., the **NER task**.
According to the documentation I filled the config as below (I substituted confidential parts, e.g., URL and label names).
However, even though I set in addition the environment variables OPENAI_API_KEY and AZURE_OPENAI_KEY in the notebook and Linux, I get a 
**HTTPError: 404 Client Error: Resource Not Found for url**: https://{resource}.openai.azure.com/openai/deployments/{deploymentname}/chat?api-version=2023-05-15
error. 
That somehow is the same if one would access this link directly via CURL without token which is blocked of course. However, the API key is definitely provided and I use it as well successfully in LangChain usually (Chat models and Embeddings).

Does someone have successfully used the NER task with Azure API and has a proper example of the config and setup so I could compare what is maybe wrong?

I use **spacy-llm version: 0.6.2**

```
import os
from spacy_llm.util import assemble

os.environ['OPENAI_API_KEY'] = '...'
os.environ['AZURE_OPENAI_KEY'] = "..."

nlp = assemble("llm_ner.cfg")
```

llm_ner.cfg:
----------------------------
```
[nlp]
lang = "en"
pipeline = ["llm"]

[components]

[components.llm]
factory = "llm"

[components.llm.task]
@llm_tasks = "spacy.NER.v3"
labels = ["LABEL1", "LABEL2", "LABEL3"]
description = "..."

[components.llm.task.label_definitions]
...

[components.llm.model]
@llm_models = "spacy.Azure.v1"
name = "{deployment_name}"
base_url = "https://{resource}.openai.azure.com/"
api_version = "2023-05-15" 
model_type = "chat" 
config = {"temperature": 1.0}
```
----------------------------------

**Comment by rmitsch** [comment, marked as answer]
Hi @orthd, thanks for reporting this. This is a known bug and was already fixed (see https://github.com/explosion/spacy-llm/issues/336), but hasn't been released as a patch yet. v0.6.3 will include the fix for this and be released this week.

**Reply by orthd** [reply]
@rmitsch , thank you very much and sorry that I didn't see the already known bug. I'm looking forward for the next release and to start working with!

**Reply by rmitsch** [reply]
No worries, thanks for bringing it up :slightly_smiling_face: The release should be out tomorrow.
